allennlp/training/trainer.py

```diff
@@ -563,12 +563,15 @@ def _parameter_and_gradient_statistics_to_tensorboard(self, # pylint: disable=in
                     grad_data = param.grad.data._values()
                 else:
                     grad_data = param.grad.data
-                self._tensorboard.add_train_scalar("gradient_mean/" + name,
-                                                   grad_data.mean(),
-                                                   epoch)
-                self._tensorboard.add_train_scalar("gradient_std/" + name,
-                                                   grad_data.std(),
-                                                   epoch)
+
+                # skip empty gradients
```

Comment:
I'm not sure this is a good idea? I definitely want to know if some of my parameters have zero gradients.

Reply:
This is just for logging to tensorboard, and is different from zero gradients.  It's the case where there a parameter with requires_grad=True, but the gradient isn't computed when the parameter isn't used in computing the loss.  This is the case where `param.grad=None` (that we exclude above for dense gradients) but in the sparse gradient case (`Embedding`modules) torch instead has a `param.grad.data` with shape of all zeros.

Reply:
Ah got it - can you add that as a comment because it's not immediately obvious that this is what the condition would apply to.